### PR TITLE
docs: document `render` options

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Document `render` options, especially the new `unstable_validateStringsRenderedWithinText` option.

### Test plan

All checks pass.

### CODE REVIEW

File: website/docs/API.md [hunk 1]
```diff
@@ -86,6 +85,40 @@ Latest `render` result is kept in [`screen`](#screen) variable that can be impor
 Using `screen` instead of destructuring `render` result is recommended approach. See [this article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen) from Kent C. Dodds for more details.
 :::
 
+### `render` options
+
+The behavior of `render` method can be customized by passing various options:
```
Comment (thymikee) [inline website/docs/API.md:90]:
```suggestion
The behavior of `render` method can be customized by passing various options as a second argument of `RenderOptions` type:
```

File: website/docs/API.md [hunk 2]
```diff
@@ -86,6 +85,40 @@ Latest `render` result is kept in [`screen`](#screen) variable that can be impor
...
+
+This options allows you to pass `createNodeMock` option to `ReactTestRenderer.create()` method in order to allow for custom mock refs. You can learn more about this options from [React Test Renderer documentation](https://reactjs.org/docs/test-renderer.html#ideas).
+
+#### `unstable_validateStringsRenderedWithinText` option
+
+```ts
+unstable_validateStringsRenderedWithinText?: boolean;
+```
+
+This experimental options allows you to replicate React Native behavior of throwing `Invariant Violation: Text strings must be rendered within a <Text> component` error when you try to render `string` value under component different than `<Text>` component, e.g. under `<View>`.
```
Comment (thymikee) [inline website/docs/API.md:114]:
```suggestion
This **experimental** options allows you to replicate React Native behavior of throwing `Invariant Violation: Text strings must be rendered within a <Text> component` error when you try to render `string` value under component different than `<Text>` component, e.g. under `<View>`. 
```

File: website/docs/API.md [hunk 3]
```diff
@@ -86,6 +85,40 @@ Latest `render` result is kept in [`screen`](#screen) variable that can be impor
...
+
+```ts
+unstable_validateStringsRenderedWithinText?: boolean;
+```
+
+This experimental options allows you to replicate React Native behavior of throwing `Invariant Violation: Text strings must be rendered within a <Text> component` error when you try to render `string` value under component different than `<Text>` component, e.g. under `<View>`. 
+
+This check is not enforced by React Test Renderer and hence by default React Native Testing Library also does not check this. That might result in runtime errors when running your code on a device, while the code works without errors in tests.
+
+:::note
```
Comment (thymikee) [inline website/docs/API.md:118]: move this note to the top of the section maybe?

**Review by thymikee — APPROVED** [review]

**Comment by mdjastrzebski** [comment]
🎉 This PR is included in version [11.2.0](https://github.com/callstack/react-native-testing-library/releases/tag/v11.2.0) 🎉

**Pull request merged**